docs/proposals/runtime-image.md

```diff
@@ -0,0 +1,122 @@
+---
+title: runtime-image-support
+authors:
+  - "@sbose78"
+  - "@otaviof"
+status: design
+---
+
+Runtime-Image Support
+---------------------
+
+# Summary
+
+This enhancement proposal describes the support of runtime-image in build-v2 operator. A runtime-image is a composition based on the image just created by a third-party builder, used as input for the runtime-image.
+
+# Motivation
+
+With the runtime-image feature in place, it will be employed for the creation of _lean-images_, which takes the ability of picking only certain parts of the original image, creating a runtime-image with the reduced data-set.
+
+Additionally, creating images with a different base-image, and other advanced use-cases are also supported by this feature.
+
+# Proposal
+
+The sections below are describing the changes necessary to introduce the runtime-image.
+
+## API Changes
+
+A new attribute will be added on Build resource, as the following example:
+
+
+```yml
+---
+apiVersion: build.dev/v1alpha1
+kind: Build
+# ...
+spec:
+  runtime:
+    base:
+      image: runtime-base-image:latest
+    workDir: /workspace/source
+    env:
+      JAVA_HOME: /path/to/java-home
+    labels:
+      custom-image-label: value
+    directories:
+      - /path/to/directory
+      - /path/to/another/directory:/target/location
+    entrypoint:
+      - java
+      - -jar
+      - ...
+```
+
+The new attributes added are:
+
+- `spec.runtime.base`: specifies the runtime base-image to be used, using [Image](https://github.com/redhat-developer/build/blob/97012ab56417ce1691a70896d90e490ea6a4d23c/pkg/apis/build/v1alpha1/build_types.go#L58) as type;
+- `spec.runtime.workDir`: path to `WORKDIR` in runtime-image;
+- `spec.runtime.env`: runtime-image additional environment variables, key-value;
+- `spec.runtime.labels`: runtime-image additional labels, key-value;
+- `spec.runtime.directories`: list of directories paths to be copied to runtime-image, those can be defined as `<source>:<destination>` split by colon (`:`);
```

Comment:
One can also copy individual files using that mechanism and not only directories. Only neutral alternative that comes to my mind right away is `paths`.

Reply:
Good point, `paths` is better wording to this attribute. Thanks!